core/src/main/java/com/linecorp/armeria/common/ContentDispositionBuilder.java

```diff
@@ -0,0 +1,169 @@
+        requireNonNull(name, "name");
+        this.name = name;
+        return this;
+    }
+
+    /**
+     * Sets the value of the {@code filename} parameter. The given
+     * filename will be formatted as quoted-string, as defined in RFC 2616,
+     * section 2.2, and any quote characters within the filename value will
+     * be escaped with a backslash, e.g. {@code "foo\"bar.txt"} becomes
+     * {@code "foo\\\"bar.txt"}.
+     */
+    public ContentDispositionBuilder filename(String filename) {
+        return filename(filename, null);
+    }
+
+    /**
+     * Sets the value of the {@code filename*} that will be encoded as defined in the RFC 5987.
+     * Only the US-ASCII, UTF-8 and ISO-8859-1 charsets are supported.
+     *
+     * <p><strong>Note:</strong> Do not use this for a {@code "multipart/form-data"} requests as per
+     * <a link="https://tools.ietf.org/html/rfc7578#section-4.2">RFC 7578, Section 4.2</a>
+     * and also RFC 5987 itself mentions it does not apply to multipart requests.
+     */
+    public ContentDispositionBuilder filename(String filename, @Nullable Charset charset) {
+        requireNonNull(filename, "filename");
+        checkArgument(!filename.isEmpty(), "filename should not be empty.");
+        this.filename = filename;
+        this.charset = charset;
+        return this;
+    }
+
+    /**
+     * Set the value of the {@code size} parameter.
+     *
+     * @deprecated As per <a href="https://tools.ietf.org/html/rfc6266#appendix-B">RFC 6266, Appendix B</a>,
+     *             to be removed in a future release.
+     */
+    @Deprecated
+    public ContentDispositionBuilder size(long size) {
+        this.size = size;
+        return this;
+    }
+
+    /**
+     * Set the value of the {@code creation-date} parameter.
+     *
+     * @deprecated As per <a href="https://tools.ietf.org/html/rfc6266#appendix-B">RFC 6266, Appendix B</a>,
+     *             to be removed in a future release.
+     */
+    @Deprecated
+    public ContentDispositionBuilder creationDate(ZonedDateTime creationDate) {
+        requireNonNull(creationDate, "creationDate");
+        this.creationDate = creationDate;
+        return this;
+    }
+
+    /**
+     * Set the value of the {@code modification-date} parameter.
+     *
+     * @deprecated As per <a href="https://tools.ietf.org/html/rfc6266#appendix-B">RFC 6266, Appendix B</a>,
+     *             to be removed in a future release.
+     */
+    @Deprecated
+    public ContentDispositionBuilder modificationDate(ZonedDateTime modificationDate) {
+        requireNonNull(modificationDate, "modificationDate");
+        this.modificationDate = modificationDate;
+        return this;
+    }
+
+    /**
+     * Set the value of the {@literal read-date} parameter.
+     * @deprecated As per <a href="https://tools.ietf.org/html/rfc6266#appendix-B">RFC 6266, Appendix B</a>,
+     *             to be removed in a future release.
+     */
+    @Deprecated
```

Comment:
Perhaps we could remove all these deprecated methods because the majority of user agents do not implement these?

Reply:
I kept these deprecated methods just in case.
I agree to remove these methods. Any thoughts @trustin?

Reply:
Let me search the usage of these fields on GitHub.

Reply:
The `creation-date` and `modification-date` seem to be only used in email MIME. Otherwise, it is rarely used. I will remove the deprecated fields.

Reply:
OK.